src/Our.Umbraco.InnerContent/Web/UI/App_Plugins/InnerContent/js/innercontent.js

```diff
@@ -261,26 +264,42 @@ angular.module("umbraco.directives").directive("innerContentOverlay", [
             };
 
             scope.openContentEditorOverlay = function () {
+                setOverlayClasses(scope.config.propertyAlias, scope.currentItem.contentTypeAlias);
                 scope.contentEditorOverlay.title = "Edit " + scope.currentItem.contentTypeName;
                 scope.contentEditorOverlay.dialogData = { item: scope.currentItem };
                 scope.contentEditorOverlay.show = true;
             };
 
             scope.closeContentEditorOverlay = function () {
+                scope.overlayClasses = [];
                 scope.contentEditorOverlay.show = false;
             };
 
             scope.closeAllOverlays = function () {
+                scope.overlayClasses = [];
                 scope.closeContentTypePickerOverlay();
                 scope.closeContentEditorOverlay();
                 scope.config.show = false;
             };
 
+            function setOverlayClasses() {
+                // Ensures that the "create" and "edit" classes don't conflict.
+                if (scope.overlayClasses.length > 1) {
+                    scope.overlayClasses.length = 1;
```

Comment:
Are you cropping the array to only 1 class? Is there potential for this to go wrong? ie, the class isn't where you expect it etc?

Reply:
When going down the Create pathway, the "create" class is added, then when a doctype is selected, the other classes are added, but the "create" one still exists, so I reduce it down to 1 (to keep the "overlay0" class).